src/core_modules/capture-core/components/LockedSelector/QuickSelector/ActionButtons.component.js

```diff
@@ -26,13 +26,41 @@ type Props = $ReadOnly<{|
     onStartAgainClick: () => void,
     onNewClick: () => void,
     onFindClick: () => void,
+    onFindClickWithoutProgramId: () => void,
     showResetButton: boolean,
 |}>;
 
+const programTypes = {
+    TRACKER_PROGRAM: 'TRACKER_PROGRAM',
+    EVENT_PROGRAM: 'EVENT_PROGRAM',
+};
+
+function useProgramInfo(programId) {
+    const [programName, setName] = useState('');
+    const [programType, setType] = useState('');
+    const [trackedEntityName, setTrackedEntityName] = useState('');
+
+    useEffect(() => {
```

Comment:
Why do we use `useEffect` here and why are we saving state here? Wouldn't `useMemo` make more sense?

Reply:
Yeah good point. Do you know any differences in those two concepts?

Reply:
`useEffects` is for side effect and runs after every render. More here: [useEffect](https://reactjs.org/docs/hooks-effect.html)